app/controllers/employments/UpdateEmploymentController.scala

```diff
@@ -160,86 +160,90 @@ class UpdateEmploymentController @Inject() (
+              journeyCacheNewRepository.set(
+                request.userAnswers
+                  .setOrException(
+                    UpdateEmploymentTelephoneQuestionPage,
+                    Messages(s"tai.label.${form.yesNoChoice.getOrElse(FormValuesConstants.NoValue).toLowerCase}")
+                  )
+                  .setOrException(UpdateEmploymentTelephoneNumberPage, form.yesNoTextEntry.getOrElse(""))
+              )
+            case _ =>
+              journeyCacheNewRepository.set(
+                request.userAnswers
+                  .setOrException(
+                    UpdateEmploymentTelephoneQuestionPage,
+                    Messages(s"tai.label.${form.yesNoChoice.getOrElse(FormValuesConstants.NoValue).toLowerCase}")
+                  )
+                  .setOrException(UpdateEmploymentTelephoneNumberPage, "")
+              )
           }
-          journeyCacheService.cache(dataForCache) map { _ =>
+          yesNoTask.map(_ =>
             Redirect(controllers.employments.routes.UpdateEmploymentController.updateEmploymentCheckYourAnswers())
-          }
+          )
         }
       )
   }
 
-  def updateEmploymentCheckYourAnswers(): Action[AnyContent] = authenticate.authWithValidatePerson.async {
+  def updateEmploymentCheckYourAnswers(): Action[AnyContent] = authenticate.authWithDataRetrieval.async {
     implicit request =>
       implicit val user: AuthedUser = request.taiUser
+      val employmentId = request.userAnswers.get(UpdateEmploymentIdPage)
+      val employmentName = request.userAnswers.get(UpdateEmploymentNamePage)
+      val employmentDetails = request.userAnswers.get(UpdateEmploymentDetailsPage)
+      val employmentTelephoneQuestion = request.userAnswers.get(UpdateEmploymentTelephoneQuestionPage)
+      val employmentTelephoneNumber = request.userAnswers.get(UpdateEmploymentTelephoneNumberPage)
 
-      journeyCacheService
-        .collectedJourneyValues(
-          Seq(
-            UpdateEmploymentConstants.EmploymentIdKey,
-            UpdateEmploymentConstants.NameKey,
-            UpdateEmploymentConstants.EmploymentDetailsKey,
-            UpdateEmploymentConstants.TelephoneQuestionKey
-          ),
-          Seq(UpdateEmploymentConstants.TelephoneNumberKey)
-        )
-        .map {
-          case Right((mandatoryJourneyValues, optionalSeq)) =>
+      (employmentId, employmentName, employmentDetails, employmentTelephoneQuestion, employmentTelephoneNumber) match {
+
+        case (Some(_), Some(_), Some(_), Some(_), _) =>
+          Future.successful(
             Ok(
               updateEmploymentCheckYourAnswers(
                 UpdateEmploymentCheckYourAnswersViewModel(
-                  mandatoryJourneyValues.head.toInt,
-                  mandatoryJourneyValues(1),
-                  mandatoryJourneyValues(2),
-                  mandatoryJourneyValues(3),
-                  optionalSeq.head
+                  employmentId.get,
```

Comment:
Don't use .gets here. Instead of using _ in case match at lines 209 say case (Some(empId), ...) and use these